projects/js-packages/publicize-components/src/components/social-image-generator/template-picker/index.js

```diff
@@ -1,80 +1,58 @@
+ * The pure template picker component. Does not save the template changes, just sends it back to the parent component,
+ * with the onTemplateSelected callback.
+ *
+ * @param {{value: string|null, onTemplateSelected: Function}} props - The component props:
+ * Value is the name of the currently selected template, onTemplateSelected is a function that
+ * will be called when a template is selected. Receives the name of the selected template as an argument.
  * @returns {React.ReactNode} - The component's rendered output.
  */
-const TemplatePicker = ( { onSelect, render, value = null } ) => {
-	const [ isOpen, setIsOpen ] = useState( false );
+const TemplatePicker = ( { value = null, onTemplateSelected = null } ) => {
 	const [ selectedTemplate, setSelectedTemplate ] = useState( value );
 
-	const openPicker = useCallback( () => setIsOpen( true ), [ setIsOpen ] );
-	const closePicker = useCallback( () => {
-		setSelectedTemplate( value );
-		setIsOpen( false );
-	}, [ value, setSelectedTemplate, setIsOpen ] );
-	const saveAndClosePicker = useCallback( () => {
-		onSelect( selectedTemplate );
-		setIsOpen( false );
-	}, [ onSelect, setIsOpen, selectedTemplate ] );
-
 	// Add memoized callback function for each template.
 	const TEMPLATES = useMemo(
 		() =>
 			TEMPLATES_DATA.map( template => ( {
 				...template,
-				onSelect: () => setSelectedTemplate( template.name ),
+				onSelect: () => {
+					setSelectedTemplate( template.name );
+					if ( onTemplateSelected ) {
+						onTemplateSelected( template.name );
+					}
+				},
 			} ) ),
-		[ setSelectedTemplate ]
+		[ setSelectedTemplate, onTemplateSelected ]
 	);
 
 	return (
-		<ThemeProvider targetDom={ document.body }>
-			{ render( { open: openPicker } ) }
-			{ isOpen && (
-				<Modal onRequestClose={ closePicker } title={ __( 'Pick a Template', 'jetpack' ) }>
-					<div className={ styles.templates }>
-						{ TEMPLATES.map( template => (
-							<button
-								onClick={ template.onSelect }
-								key={ template.name }
-								className={ classnames( styles.template, {
-									[ styles[ 'template--active' ] ]: template.name === selectedTemplate,
-								} ) }
-							>
-								<img src={ template.image } alt={ template.label } />
-								<span className="screen-reader-text">
-									{
-										/* translators: %s is the name of the template */
-										sprintf( __( 'Pick the %s template', 'jetpack' ), template.label )
-									}
-								</span>
-							</button>
-						) ) }
-					</div>
-					<div className={ styles.footer }>
-						<Button variant="tertiary" onClick={ closePicker }>
-							{ __( 'Cancel', 'jetpack' ) }
-						</Button>
-						<Button variant="primary" onClick={ saveAndClosePicker }>
-							{ __( 'Save', 'jetpack' ) }
-						</Button>
-					</div>
-				</Modal>
-			) }
-		</ThemeProvider>
+		<div className={ styles.templates }>
+			{ TEMPLATES.map( template => (
+				<button
+					onClick={ template.onSelect }
```

Comment:
We might even be able to define the callback here then, so we don't have to set `onSelect` on each of the templates first.